File: Source/Aladdin.CAPI.ANSI.Java/src/aladdin/capi/ansi/x942/PrivateKey.java
package aladdin.capi.ansi.x942;
import aladdin.capi.*; 
import java.math.*; 

///////////////////////////////////////////////////////////////////////////
// Личный ключ алгоритма DH
///////////////////////////////////////////////////////////////////////////
public class PrivateKey extends aladdin.capi.PrivateKey implements IPrivateKey
{
    // номер версии для сериализации
    private static final long serialVersionUID = -6392617364637658603L;
    
    // параметры ключа и секретное значение
    private final IParameters parameters; private final BigInteger x;
    
    // конструктор
	public PrivateKey(Factory factory, SecurityObject scope, 
        String keyOID, IParameters parameters, BigInteger x) 
	{ 	
        // сохранить переданные параметры
		super(factory, scope, keyOID); this.parameters = parameters; this.x = x; 
    } 
    // параметры ключа
	@Override public final IParameters parameters() { return parameters; } 
    // секретное значение
	@Override public final BigInteger getX() { return x; } 
}

File: Source/Aladdin.CAPI.ANSI/CAPI/ANSI/X942/PrivateKey.cs
using System;
using System.Diagnostics.CodeAnalysis;

namespace Aladdin.CAPI.ANSI.X942
{
	///////////////////////////////////////////////////////////////////////////
	// Личный ключ алгоритма DH
	///////////////////////////////////////////////////////////////////////////
    [SuppressMessage("Microsoft.Design", "CA1063:ImplementIDisposableCorrectly")]
	public class PrivateKey : CAPI.PrivateKey, IPrivateKey
	{
        // параметры ключа и секретное значение
        private IParameters parameters; private Math.BigInteger x;
    
        // конструктор
	    public PrivateKey(CAPI.Factory factory, SecurityObject scope, string keyOID, 
            IParameters parameters, Math.BigInteger x) : base(factory, scope, keyOID)
	    { 	
            // сохранить переданные параметры
		    this.parameters = parameters; this.x = x; 
        } 
        // параметры ключа
	    public override CAPI.IParameters Parameters { get { return parameters; }}
        // секретное значение
	    public virtual Math.BigInteger X { get { return x; }}
    }
}
